packages/job-manager/src/batch/batch.ts

```diff
@@ -26,6 +28,61 @@ export class Batch {
         @inject(Logger) private readonly logger: Logger,
     ) {}
 
+    public async getPoolTaskProcessingRatio(): Promise<number> {
+        const resultVariableName = 'taskProcessingRatioResult';
+        const poolTaskProcessingRatioFormula = `
+        // The script will calculate the tasks processing ratio returned in $taskProcessingRatio variable as
+        // double value and will return -1 when there are no active tasks or sampling data is insufficient
+
+        // the time interval for data sampling
+        $sampleTimeIntervalMinutes = 15 * TimeInterval_Minute;
+        $samplePercentThreshold = 70;
+
+        // get sample data for pending tasks (the sum of active and running tasks)
```

Comment:
why not call batch apis to know number of idle nodes & number of running nodes instead of using the auto scale formula?